sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -877,6 +879,30 @@ public CosmosClientBuilder sessionRetryOptions(SessionRetryOptions sessionRetryO
         return this;
     }
 
+    /**
+     * Sets the regions to exclude from the list of preferred regions. This means the request will not be
+     * routed to these excluded regions for non-retry and retry scenarios
+     * for the workload executed through this instance of {@link CosmosClient} / {@link CosmosAsyncClient}.
+     *
+     * @param excludedRegions The list of regions to exclude.
+     * @return current CosmosClientBuilder.
+     * */
+    public CosmosClientBuilder excludeRegions(List<String> excludedRegions) {
```

Comment:
What if customer passes an empty list here?

Reply:
There won't be regions to exclude from `preferredRegions` typically. It is a valid use case.